### PR TITLE
weapons-recoil

### PR DESCRIPTION
**Pull request by SmallPeenwow**
Sorry it's done just what to do the merge conflict and do some more calculations and some other stuff to get it perfect but getting brain fart and I need to think again what all else to do.

Is not done and will let you know when done.

### CODE REVIEW

**Comment by SmallPeenwow** [comment]
Going to do conflicts and other stuff

**Comment by SmallPeenwow** [comment]
@itbeginshere  I'm not quite done i think but I just want you to test some things out for me while I have loadshedding. Please look at docs also.

For the test I want you to do look at M4 Carbine and AR15 Sig M400 VTAC. At top of assault Rifle list.

There is code commented out that I would like you to test around with the smaller values on the 2 guns recoil upwards in the () those things has testing next to it.

As it would increase the recoil affective if full auto or if you okay with the bigger values and just want them to be incremented in smaller amounts for the full auto affect

recoil velocity is recoil upwards and recoil horizontal is half of recoil upwards.

All is docs besides for the commented out stuff since it was a tester to see but also taking into fact about gun weight and recoil impulse or recoil energy. 

Look under Recoil Calculator sites in docs and use the site to check what recoil impulse does and recoil energy or just a small understanding by looking at formula otherwise just do the small test and let me know if you want the recoil to be less and increase or big and increase little when full auto


yesyes

**Comment by SmallPeenwow** [comment]
Sorry Not DONE Yet

**Comment by SmallPeenwow** [comment]
@itbeginshere Its done now. Just check voice notes first.

Could get that one thing with the semi-auto vertical height restrictions but you give it a test and see if you think it will matter.

Since i'm sure the player would of tried to pull down or just aim down sight but it doesn't seem to affect much unless you not going to pull down on the mouse.

File: paranomal-game/Assets/Scripts/Player/Recoil.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+using UnityEngine;
+
+public class Recoil : MonoBehaviour
+{
+    // Rotations
+    private Vector3 currentRotation;
+    private Vector3 targetRotation;
```
Comment (itbeginshere) [inline paranomal-game/Assets/Scripts/Player/Recoil.cs:7]: Maybe some comments for these ones?

File: paranomal-game/Assets/Scripts/Player/Recoil.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        currentRotation = Vector3.Slerp(currentRotation, targetRotation, snappiness * Time.deltaTime);
+        
+        transform.localRotation = Quaternion.Euler(currentRotation);
+    }
+
+    public void RecoilFire()
+    {
+        // weapon.verticalRecoil will move the weapon upwards on the X axis with the negative sign
+        // weapon.horizontalRecoil will move the weapon along the sides of the Y axis
+        if(weapon.isFullAuto && playerPrefebInputManger.GetComponent<WeaponSystem>().triggerDown) // Posible don't use
```
Comment (itbeginshere) [inline paranomal-game/Assets/Scripts/Player/Recoil.cs:54]: Possible don't use

File: paranomal-game/Assets/Scripts/Player/WeaponSystem.cs [hunk 3]
```diff
@@ -6,13 +6,21 @@ public class WeaponSystem : MonoBehaviour
...
+
     // Update is called once per frame
     void Update()
     {
         if (rightHand.transform.childCount != 0)
         {
             rightHandWeapon = rightHand.GetComponentInChildren<Weapon>(); // Gets RightHand GameObject
-        }
+
+            if (triggerDown && rightHandWeapon.isFullAuto)
```
Comment (itbeginshere) [inline paranomal-game/Assets/Scripts/Player/WeaponSystem.cs:19]: Indenting seems off here

**Review by itbeginshere — COMMENTED** [review]
-

File: paranomal-game/Assets/Scripts/Player/Recoil.cs (hunk 2 above)
Comment (SmallPeenwow) [reply paranomal-game/Assets/Scripts/Player/Recoil.cs:54]: what must I not use?

Comment (SmallPeenwow) [reply paranomal-game/Assets/Scripts/Player/Recoil.cs:54]: Oh i see what you mean. I think it was just something I left there before I did other stuff and thinking and forgot about it

**Comment by SmallPeenwow** [comment]
Its all done now @itbeginshere.

**Pull request merged**